shell/common/gin_converters/usb_device_info_converter.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2022 Microsoft, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ELECTRON_SHELL_COMMON_GIN_CONVERTERS_USB_DEVICE_INFO_CONVERTER_H_
+#define ELECTRON_SHELL_COMMON_GIN_CONVERTERS_USB_DEVICE_INFO_CONVERTER_H_
+
+#include "gin/converter.h"
+#include "services/device/public/mojom/usb_device.mojom.h"
+#include "shell/browser/usb/usb_chooser_context.h"
+#include "shell/common/gin_converters/value_converter.h"
+
+namespace gin {
+
+template <>
+struct Converter<device::mojom::UsbDeviceInfoPtr> {
+  static v8::Local<v8::Value> ToV8(
+      v8::Isolate* isolate,
+      const device::mojom::UsbDeviceInfoPtr& device) {
+    base::Value value = electron::UsbChooserContext::DeviceInfoToValue(*device);
```

Comment:
The converter can take `UsbDeviceInfo`  instead of the `UsbDeviceInfoPtr` as parameter, so the code can do `Emit("usb-device-event", device_info)` instead of `Emit("usb-device-event", device_info.Clone())` to save a clone.

```c++
 template <>
 struct Converter<device::mojom::UsbDeviceInfo> {
   static v8::Local<v8::Value> ToV8(
       v8::Isolate* isolate,
       const device::mojom::UsbDeviceInfo& device) {
     base::Value value = electron::UsbChooserContext::DeviceInfoToValue(device);
     return gin::ConvertToV8(isolate, value);
   }
 };
```

Reply:
The converter is also used for the `select-usb-device` event which passes along a vector of  `device::mojom::UsbDeviceInfoPtr`s